### ISSUE TITLE
Trojan Horse in Updater.exe?

### ISSUE THREAD

**Issue by damiandib** [Rem0o/FanControl.Releases#719]
It might be a fluke from Windows Defender but it claims it detected a Trojan Horse in the Updater.exe, and I saw the Updater.exe was just updated so it seemed like an odd coincidence 
I'll be happy to provide more information if needed
![afbeelding](https://user-images.githubusercontent.com/12232680/166115674-fa236ac8-6646-4d86-8c6a-5c60079f7fa8.png)

**Comment by f8lxninja** [comment]
Just experienced this as well.

**Comment by Rem0o** [comment]
Updated the app manifest to explicitly use the same invocation privilege as the app starting it, aka FanControl.exe. 

It's a .NET exe, feel free to decompile it, nothing to hide.

**Comment by damiandib** [comment]
@Rem0o I am sorry if this seemed like an attack, that's not what I am trying to achieve here at all! I am just trying to make you aware of this issue I have. I am not trying to say that you put a Trojan Horse there on purpose or at all...

**Comment by KingM00se** [comment]
Yeah, just had the exact same warning come up after attempting to install todays update. Hmm...

**Comment by arjenkrap** [comment]
A makeshift alternative to Updater.exe using git. (You do need to have git and 7zip installed.)

First clone the FanControl.Releases repo:

`git.exe clone https://github.com/Rem0o/FanControl.Releases.git`

Add  a file update.bat
```
@echo off
taskkill.exe /f /im FanControl.exe
git.exe pull
7z.exe x -aoa FanControl.zip
start FanControl.exe
```

Add a file .gitignore (optional)
```
*.dll
CACHE
FanControl.exe
FanControl.exe.config
FanControl.sys
userConfig.json
```

Now run update.bat to update:
```
SUCCESS: The process "FanControl.exe" with PID 6656 has been terminated.
Already up to date.

7-Zip 21.07 (x64) : Copyright (c) 1999-2021 Igor Pavlov : 2021-12-26

Scanning the drive for archives:
1 file, 4952378 bytes (4837 KiB)

Extracting archive: FanControl.zip
--
Path = FanControl.zip
Type = zip
Physical Size = 4952378

Everything is Ok

Files: 23
Size:       15100846
Compressed: 4952378
```

Optionally commit update.bat (or add it to .gitignore):

```
git.exe add update.bat
git.exe commit - "Added update.bat"
```

**Comment by cyounkins** [comment]
Here's virustotal for Updater.exe: https://www.virustotal.com/gui/file/52db36eac73eac1c220ca1ec29f674ac822e2a30839e2a540f6b6ecb1a49a431/detection

It shows that the Updater is pulling FanControl.zip through a bit.ly redirect link. Can you explain why that is done?

**Comment by BinaryAssault** [comment]
> Here's virustotal for Updater.exe: https://www.virustotal.com/gui/file/52db36eac73eac1c220ca1ec29f674ac822e2a30839e2a540f6b6ecb1a49a431/detection
> 
> It shows that the Updater is pulling FanControl.zip through a bit.ly redirect link. Can you explain why that is done?

Not advocating for the dev here, but that bit.ly points back to github. Not sure why it's done that way here. You can preview a bit.ly link by adding a '+' to the end so bit.ly/abcd would be bit.ly/abcd+ 
doing so will tell you the full file path via bit.ly itself without visiting the page ([source](https://support.bitly.com/hc/en-us/articles/230650447-Can-I-preview-a-link-before-clicking-on-it-))

That link in particular in virus total reports:
https://raw.githubusercontent.com/Rem0o/FanControl.Releases/master/FanControl.zip

**Comment by Rem0o** [comment]
@cyounkins  I used the bitly link as a simple updater/download counter.
![image](https://user-images.githubusercontent.com/14118956/166130351-16ae517d-3434-4dab-8bc7-b7c01da903e9.png)

**Comment by Rem0o** [comment]
To be honest, everytime I make a change to that exe, it triggers an anti-virus software somewhere and I have to do a new submission for false-positive. I might change it to a cmd script or something to never have to deal with these AV freaking out everytime.

**Comment by Rem0o** [comment]
Reverted back to the previous updater for now before I find a better solution going forwards.

**Comment by BinaryAssault** [comment]
> To be honest, everytime I make a change to that exe, it triggers an anti-virus software somewhere and I have to do a new submission for false-positive. I might change it to a cmd script or something to never have to deal with these AV freaking out everytime.

I decompiled the exe and there isn't anything malicious I can see in there. 

Have you tried [signing the code](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-sign-application-and-deployment-manifests?view=vs-2022)?

**Comment by Rem0o** [comment]
@BinaryAssault 
Wanted to avoid dealing with that, but I guess we're at that point.  Won't a need a paid certificate from a third-party though?

**Comment by cyounkins** [comment]
After searching more, I am inclined to believe this is a false positive. Github has 255 issues open with "Wacatac.B!ml" including gems such as compiling "hello world" in rust (https://github.com/rust-lang/rust/issues/88297) and Google's Project Zero tools (https://github.com/googleprojectzero/sandbox-attacksurface-analysis-tools/issues/54).

Binary signing may help, not sure. Others report that twiddling the linker flags changes detection.

**Comment by WantStuff** [comment]
I just got the same "Trojan" quarantine with the latest update, v137

**Comment by damiandib** [comment]
I had this myself again today as well

**Comment by Revivedx** [comment]
Just had this message 3 minutes ago.

**Comment by Rem0o** [comment]
Submitted for review already. Waiting on the whitelisting.

**Comment by Scrachy** [comment]
@Rem0o means our / mine todo is waiting?

**Comment by Globespy** [comment]
Good, then I should tell Windows that it doesn't need to quarantine 'updater.exe'.
Hope you get it figured out soon, I'm sure everyone here is just making you aware of this false positive, which is better than panicking, uninstalling and not saying a thing.
Your program is the best fan control app I have ever used, I avoid using everything else - especially love the 'Mix' feature, haven't seen that before in other programs.
Keep up the great work Reme!

**Comment by TheEngineerGuy** [comment]
Trojan Warning is back! A year later...

**Comment by caliwator** [comment]
Yep, got the same warning today.

**Comment by AdamAnon** [comment]
Hi, yes same here. tried to update FanControl and got a warning from Windows Security about a Trojan Win32/Wacatac.B!ml. So was it ever whitelisted? Thanks!

**Comment by jaxteri** [comment]
I just got it as well, I added an exception Defender but the update seems to have failed and FanControl does not start at all anymore...

**Comment by Killrockstar** [comment]
Also just got it

**Comment by Dex1975** [comment]
![error](https://github.com/Rem0o/FanControl.Releases/assets/44803658/fc2f3acc-79ee-46b0-8fdc-519688b3d438)

On clicking update, endless "Downloading Updater". click cancel and I get the screen.

**Comment by AdamAnon** [comment]
> I just got it as well, I added an exception in Defender but the update seems to have failed and FanControl does not start at all anymore...

So I've downloaded the new version 174 manually and copied the \Configurations directory from the old folder and it works.

**Comment by SeeUnsharp** [comment]
I get the same.
![grafik](https://github.com/Rem0o/FanControl.Releases/assets/20801755/b0244c72-cf95-48dd-939c-bff99685f25d)

**Comment by joseseat** [comment]
Me too. Is this something to be concerned about or just an error?? I have quarantined the Updater.exe...

**Comment by ash-j-f** [comment]
![image](https://github.com/Rem0o/FanControl.Releases/assets/32670720/5ea66882-7fed-46ab-94d1-f42c9953399e)
Just another report, fwiw! Trying to go from 173 to 174 Thanks for the hard work

**Comment by Skripeezy** [comment]
![image](https://github.com/Rem0o/FanControl.Releases/assets/149828686/7e6c944c-a591-4cb1-9bc5-ac3cee4f160e)

I saw what people said, but I let it through my firewall anyway. But if that trojan is positive, it can mess up a pc big time. It will make copies of every file until the system freezes for good

**Comment by Skripeezy** [comment]
> ![image](https://user-images.githubusercontent.com/149828686/284910000-7e6c944c-a591-4cb1-9bc5-ac3cee4f160e.png)
> 
> I saw what people said, but I let it through my firewall anyway. But if that trojan is positive, it can mess up a pc big time. It will make copies of every file until the system freezes for good

thank you for being an active developer with the best fan control and most reliable out there homie!!!!!

**Comment by Rem0o** [comment]
New updater, same problem.

Duplicate of  https://github.com/Rem0o/FanControl.Releases/issues/2135

**Comment by mattj04** [comment]
I know this is probably already been covered, but I would like to add that virustotal also flags the ring0 in the .sys. Dr Web, ArcticWolf, and Rising all flagged it as low MITRE score driver vulnerability, the latter two specifically as "Tool.VulnDriver.23" and I will link the DrWeb sandbox report. Looks like it is just what dev says, a vulnerability but not necessarily malware. So this may explain why everyone's WinDefender keeps flagging it. It's just telling us FanControl could be malware if it wanted to, not that it actually is malware. 
[VirusTotal Report](https://www.virustotal.com/gui/file/11bd2c9f9e2397c9a16e0990e4ed2cf0679498fe0fd418a3dfdac60b5c160ee5/behavior)
[vxCube Report](https://vtbehaviour.commondatastorage.googleapis.com/11bd2c9f9e2397c9a16e0990e4ed2cf0679498fe0fd418a3dfdac60b5c160ee5_Dr.Web%20vxCube.html?GoogleAccessId=758681729565-rc7fgq07icj8c9dm2gi34a4cckv235v1@developer.gserviceaccount.com&Expires=1756002253&Signature=OatarTgphhFHmbEyIFF4%2BCk08jh1SMHGumnu7kqdPjkTCPOfdQWQ94%2FeQX%2F4IDWMTK3g5ZFx6teDJPYq7y6Z0kS77SK8u0wr3QyxA8C4rlsg7fs3HWeULSZ4Lbf3Do0KBX58uMlF%2BjL%2F%2F6WxxogE7nc4cCDMaSH%2BRpoKr8CIMICse2cuTHOo4Gsl%2FaX3lfGpqi4EbYkgiV%2F8J8wW0rM6%2BROySidlqF4%2FELm7gZTrloFrQ%2BnRn%2FhVAtCDxKcrDiAiEVx1b%2B8mv5L%2FRePU5OGtmzNW3FEUuPZPcZVslozac9SSaU%2FQk9QzIkMBlxl6wm5GutF%2BxF4iRnyXEI8wIilVDQ%3D%3D&response-content-type=text%2Fhtml;)

**Comment by Skripeezy** [comment]
Thank you for clarifying. I already looked into it and knew but was scared
at first
